### ISSUE TITLE
Collapse PCR duplicates for Nanopore long reads data

### ISSUE THREAD

**Issue by chunjie-sam-liu** [chaolinzhanglab/ctk#5]
Hello,

I follow the https://zhanglab.c2b2.columbia.edu/index.php/Standard/BrdU-CLIP_data_analysis_using_CTK to process my long reads data.

When Gettiing unique CLIP tags by collapsing PCR duplicates, I used tag2collapse.pl to get tag.unique.bed file. It returns error as below.

```
CMD = tag2collapse.pl -big -v --random-barcode -EM 30 --seq-error-model alignment -weight --weight-in-name --keep-max-score --keep-tag-name aligned.sorted.tag.bed aligned.sorted.tag.uniq.bed
reading tags from aligned.sorted.tag.bed ...
100000 ...
100000 ...
200000 ...
300000 ...
400000 ...
500000 ...
600000 ...
700000 ...
800000 ...
900000 ...
1000000 ...
1100000 ...
.
.
.
.
Argument "b95a" isn't numeric in multiplication (*) at /scr1/users/liuc9/tools/anaconda3/envs/ctk/lib/ctk/tag2collapse.pl line 992
, <$fin> line 215908.
Argument "88a0" isn't numeric in multiplication (*) at /scr1/users/liuc9/tools/anaconda3/envs/ctk/lib/ctk/tag2collapse.pl line 992
, <$fin> line 215908.
Argument "ae51" isn't numeric in multiplication (*) at /scr1/users/liuc9/tools/anaconda3/envs/ctk/lib/ctk/tag2collapse.pl line 992
, <$fin> line 215908.
Argument "a623" isn't numeric in multiplication (*) at /scr1/users/liuc9/tools/anaconda3/envs/ctk/lib/ctk/tag2collapse.pl line 992
, <$fin> line 215908.
Argument "b95a" isn't numeric in multiplication (*) at /scr1/users/liuc9/tools/anaconda3/envs/ctk/lib/ctk/tag2collapse.pl line 992
, <$fin> line 215908.

```

In this paper https://www.nature.com/articles/s41592-019-0570-0 method section, it said that they use CIMS to identify C-to-U editing on PacBio datasets. We follow this protocol but with Nanopore sequencing.

**Comment by suvrajitm** [comment]
Hello, can you make sure the file "aligned.sorted.tag.bed" obtained in the previous step (using parseAlignment.pl) was generated properly?  Can you check what proportions of the reads were mapped uniquely ?
Did you see any other issues in any of the previous ( "preprocessing" ) steps ? 
I have not encountered this specific problem before, not sure how to reproduce the error.

**Comment by chunjie-sam-liu** [comment]
Hi Suvrajit,

1. I follow the standard tutorial, the reads were mapped to the human genome by `minimap2` and then started from `samtools fillmd` step. Next, I used `parseAlignment.pl` to get **aligned.sorted.mutation.txt** and **aligned.sorted.tag.bed**. These two files were generated properly.

2. I used `samtools flagstat`, then got the reads mapping ratio 94.94%.
![image](https://user-images.githubusercontent.com/16101520/236899006-f0b8e403-837f-42cd-9afc-3e4a5cfe755f.png).

3. No, the reads are mapped to human genome by `minimap2`, then `samtools fillmd` and then `parseAlimengt.pl`. And then `tag2collapse.pl`, this step got into error.

Below is the code I used https://github.com/chunjie-sam-liu/m6a/blob/09583b11ac1f8c85c831b321e4c7e24fbb1e69fb/test/CIMS/m6a.sh

I am not sure the tag2collapse step may influence the final result. In my CIMS analysis step, I can't get correct mutation file, the error as below.
![image](https://user-images.githubusercontent.com/16101520/236904889-49632380-1d1b-4473-bec8-04c6b25a1ea3.png).

To reproduce the error, I extract my test data into a small file, and I put it here https://www.icloud.com/iclouddrive/07bT5Dkc-Uqo6TM7tvS94XZpw#07-Data.

Very appreciate for your efforts in solving this error.
